packages/did/src/didAuthorizeExtrinsics/didAuthorizeExtrinsics.spec.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * @group unit/didbuilder
+ */
+
+import { randomAsHex } from '@polkadot/util-crypto'
+
+import { BN } from '@polkadot/util'
+import { KeyringPair, SignCallback } from '@kiltprotocol/types'
+import {
+  ApiMocks,
+  createLocalDemoFullDidFromKeypair,
+  makeSigningKeyTool,
+} from '@kiltprotocol/testing'
+import { didAuthorizeExtrinsics } from './didAuthorizeExtrinsics.js'
+import { FullDidDetails } from '../DidDetails'
+
+const mockApi = ApiMocks.createAugmentedApi()
+
+jest.mock('../Did.chain.js', () => ({
+  ...jest.requireActual('../Did.chain.js'),
+  generateDidAuthenticatedTx: jest.fn().mockResolvedValue({}),
+}))
+
+describe('didAuthorizeExtrinsics', () => {
+  let keypair: KeyringPair
+  let sign: SignCallback
+  let fullDid: FullDidDetails
+
+  beforeAll(async () => {
+    ;({ keypair, sign } = makeSigningKeyTool())
+    fullDid = await createLocalDemoFullDidFromKeypair(keypair)
+  })
+
+  describe('.addSingleExtrinsic()', () => {
+    it('fails if the extrinsic does not require a DID', async () => {
+      const extrinsic = mockApi.tx.indices.claim(1)
+      await expect(async () =>
+        didAuthorizeExtrinsics({
+          did: fullDid,
+          batchFunction: mockApi.tx.utility.batchAll,
+          extrinsics: [extrinsic, extrinsic],
+          sign,
+          submitter: keypair.address,
+        })
+      ).rejects.toMatchInlineSnapshot(
+        '[DidBuilderError: Can only batch extrinsics that require a DID signature]'
+      )
+    })
+
+    it('fails if the extrinsic is a utility (batch) extrinsic containing valid extrinsics', async () => {
```

Comment:
That sounds like, it shouldn't fail, if it contains valid extrinsics? Maybe it should say something else.

Reply:
I carried it over from another test without many changes https://github.com/KILTprotocol/sdk-js/blob/013633a/packages/did/src/DidBatcher/DidBatchBuilder.spec.ts#L77-L83